emcc.py

```diff
@@ -67,7 +67,7 @@
 C_ENDINGS = C_ENDINGS + SPECIAL_ENDINGLESS_FILENAMES # consider the special endingless filenames like /dev/null to be C
 
 EXECUTABLE_ENDINGS = ('.wasm', '.html', '.js', '.mjs', '.out', '')
-DYNAMICLIB_ENDINGS = ('.dylib', '.so') # Windows .dll suffix is not included in this list, since those are never linked to directly on the command line.
+DYNAMICLIB_ENDINGS = ('.dylib', '.so', '.wasm') # Windows .dll suffix is not included in this list, since those are never linked to directly on the command line.
```

Comment:
What about using an entirely new extension such as `.wso` ?  Would that achieve the desired effect?